### PR TITLE
Adding kn CLI to main CLI tools section

### PR DESCRIPTION
**Pull request by abrennan89**
Applies for OCP 4.3, 4.4 and 4.5

### CODE REVIEW

File: cli_reference/kn-cli-tools.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+* Full support for managing all features of Knative Serving, such as services, revisions, and traffic splitting
+* Easily create and manage Knative Eventing components, such as event sources, brokers, and triggers for filtering events.
++
+[NOTE]
+====
+Knative Eventing is currently available as a Technology Preview feature of {ServerlessProductName}.
+====
+* A plugin architecture similar to that of `kubectl` plugins
+* A thin, client-specific API in golang, that enables tasks such as synchronously waiting for Knative service write operations
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:29]: I don't think the API aspect is relevant for this scope (its about end-user, not developers). Maybe we can rephrase this that `kn` adds specific support for interactive and scripted usage like waiting synchronously on the results of an operations ?

File: cli_reference/kn-cli-tools.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+{ServerlessProductName} cannot be installed using the `kn` CLI. A cluster administrator must install the {ServerlessOperatorName} and set up the Knative components, as described in the _Serverless applications_ documentation for {product-title}.
+====
+
+== Key features
+`kn` is designed to make serverless computing tasks simple and concise, and has the following key features:
+
+* Simplest way to xref:../serverless/knative_serving/getting-started-knative-services.adoc#getting-started-knative-services[deploy serverless applications] from the command line
+* Full support for managing features of Knative Serving, such as services, revisions, and traffic splitting
+// TODO: add docs about revisions
+* Easily create and manage Knative Eventing components, such as event sources, brokers, and triggers for filtering events.
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:24]: brokers are not yet here but are supposed to land eventually in kn-land

File: cli_reference/kn-cli-tools.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+[NOTE]
+====
+Knative Eventing is currently available as a Technology Preview feature of {ServerlessProductName}.
+====
+* Create xref:../serverless/knative_eventing/serverless-sinkbinding.adoc#serverless-sinkbinding[sink binding] to connect existing Kubernetes applications and Knative services
+* Flexible plugin architecture, similar to `kubectl`, that can be used for extending `kn`
+// * Easily integrate {ServerlessProductName} with OpenShift Pipelines by using `kn` in an OpenShift Pipelines task
+// TODO: Add integrations later when we have docs about this.
+* Ideal for scripting custom rollout and rollback strategies, such as canary or blue-green deployments
+// TODO: add docs about these deployment types?
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:35]: I think that makes totally sense mid-term but for now I think this is clear enough.

Maybe rephrase the sentence so that `kn` is good for scripting Knative in general, with rollout/rollback as an example ?

File: cli_reference/kn-cli-tools.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+* Easily configure xref:../serverless/knative_serving/configuring-knative-serving-autoscaling.adoc#configuring-knative-serving-autoscaling[autoscaling] parameters using intuitive options
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:36]: "intuitive options" sounds a bit vague. Mabe just skip "using intuitive options" ? Something like

> Easily configure all autoscaling parameters for Knative services

?

File: cli_reference/kn-cli-tools.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+* Build application deployments in an iterative and interactive way, which then can be exported to standard deployment YAML files
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:37]:
```suggestion
* Build application deployments in an iterative and interactive way, which then can be exported to standard resource descriptors that can be stored in a configuration management system like Git
```

Note that this feature is still experimental (maybe one could mention that, too ?)

File: cli_reference/kn-cli-tools.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+// TODO: add docs for exporting YAML usecases?
+* Adds support for interactive and scripted usage, such as waiting synchronously for the results of an operation
```
Comment (rhuss) [inline cli_reference/kn-cli-tools.adoc:39]: We already have "scripted usage" in one item above. Maybe just skip that point, its too much of a detail.

**Review by rhuss — COMMENTED** [review]
Looks good, thanks ! See my comments below for some suggestions.

File: cli_reference/kn-cli-tools.adoc (hunk 5 above)
Comment (abrennan89) [reply cli_reference/kn-cli-tools.adoc:37]: Just gonna remove this if it's experimental for now since we're GA, so we should only doc stable stuff unless it's part of the eventing TP

File: cli_reference/kn-cli-tools.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
+include::modules/serverless-document-attributes.adoc[]
+[id="kn-cli-tools"]
+= Knative CLI (`kn`) for use with OpenShift Serverless
+:context: kn-cli-tools
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+`kn` extends the functionality of the `oc` or `kubectl` tools to enable interaction with Knative components on {product-title}.
```
Comment (cardil) [inline cli_reference/kn-cli-tools.adoc:9]: I would say that `kn` don't extend functionality of `oc` or `kubectl`, rather that simplify knative related tasks. Extends in my mind means I can do all that `oc` and `kubectl` let me, and something above that.

`oc` extends functionality of `kubectl`, that's true. `kn` is different.

**Review by cardil — COMMENTED** [review]
One suggestion. Feel free to ignore it if you feel like it doesn't matter. Apart that, lgtm.

**Review by cardil — APPROVED** [review]
QE approved

File: _topic_map.yml [hunk 8]
```diff
@@ -1368,6 +1368,8 @@ Topics:
 #    File: installing-tkn
 #  - Name: Basic tkn commands
 #    File: op-tkn-reference
+- Name: Knative CLI (kn) for use with OpenShift Serverless
```
Comment (bobfuru) [inline _topic_map.yml:1371]: Can you replace name with `{ServerlessProductName}`?

File: cli_reference/kn-cli-tools.adoc [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+toc::[]
+
+`kn` enables simple interaction with Knative components on {product-title}.
+
+You can enable Knative on {product-title} by installing {ServerlessProductName}. For more information, see the documentation on xref:../serverless/serverless-getting-started.adoc#serverless-getting-started[Getting started with {ServerlessProductName}].
+
+[NOTE]
+====
+{ServerlessProductName} cannot be installed using the `kn` CLI. A cluster administrator must install the {ServerlessOperatorName} and set up the Knative components, as described in the _Serverless applications_ documentation for {product-title}.
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:15]: Can you add an xref instead of `the _Serverless applications_ documentation`?

File: cli_reference/kn-cli-tools.adoc [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+====
+
+== Key features
+`kn` is designed to make serverless computing tasks simple and concise, and has the following key features:
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:19]: I'd break up this sentence. "Key features include:"

Comment (bobfuru) [reply cli_reference/kn-cli-tools.adoc:19]: Be sure each list item ends with a period and is a complete sentence. Try to start with a verb. I'll add some suggestions inline.

File: cli_reference/kn-cli-tools.adoc [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+* Simplest way to xref:../serverless/serving-creating-managing-apps.adoc#serving-creating-managing-apps[deploy serverless applications] from the command line
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:21]: Deploy serverless applications from the command line.

File: cli_reference/kn-cli-tools.adoc [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+* Full support for managing features of Knative Serving, such as services, revisions, and traffic splitting
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:22]: Includes full support
s/traffic splitting/traffic-splitting/

File: cli_reference/kn-cli-tools.adoc [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+// TODO: add docs about revisions
+* Easily create and manage Knative Eventing components, such as event sources and triggers
++
+[NOTE]
+====
+Knative Eventing is currently available as a Technology Preview feature of {ServerlessProductName}.
+====
+* Create xref:../serverless/knative_eventing/serverless-sinkbinding.adoc#serverless-sinkbinding[sink binding] to connect existing Kubernetes applications and Knative services
+* Flexible plugin architecture, similar to `kubectl`, that can be used for extending `kn`
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:31]: Extend `kn` with flexible plug-in architecture, similar to `kubectl`.

File: cli_reference/kn-cli-tools.adoc [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+// * Easily integrate {ServerlessProductName} with OpenShift Pipelines by using `kn` in an OpenShift Pipelines task
+// TODO: Add integrations later when we have docs about this.
+// TODO: add docs about these deployment types?
+* Easily configure xref:../serverless/knative_serving/configuring-knative-serving-autoscaling.adoc#configuring-knative-serving-autoscaling[autoscaling] parameters for Knative services
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:35]: Using adverbs like "easily" and "simply" make it sound like marketing; I'd remove them where possible.

File: cli_reference/kn-cli-tools.adoc [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+* Adds support for interactive and scripted usage, such as waiting synchronously for the results of an operation, or custom rollout and rollback strategies
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:36]: I'm not really sure what "waiting synchronously" means. Can you just say "waiting for"?
s/or custom rollout/or for deploying custom rollout/

File: cli_reference/kn-cli-tools.adoc [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+
+== Installing `kn`
+
+For information about installing `kn` for use with {ServerlessProductName}, see the documentation on xref:../serverless/installing_serverless/installing-kn.adoc#installing-kn[Installing the Knative CLI (`kn`)].
+
+== Next steps
+* For information about basic `kn` workflows and completing user tasks, see the documentation on xref:../serverless/knative_cli/knative-cli.adoc#knative-cli[Getting started with Knative CLI (`kn`)].
+* For information about completing Knative Eventing related user tasks with `kn`, see the _Knative Eventing_ documentation available in the _Serverless applications_ documentation.
```
Comment (bobfuru) [inline cli_reference/kn-cli-tools.adoc:44]: Can you add the serverless/knative_eventing/serverless-knative-eventing.adoc ref here? It looks odd having xrefs in the previous two bullets and not this one.

**Review by bobfuru — COMMENTED** [review]
Some style and formatting suggestions. :)

File: cli_reference/kn-cli-tools.adoc (hunk 15 above)
Comment (abrennan89) [reply cli_reference/kn-cli-tools.adoc:36]: @rhuss any ideas about this one? I'm not sure what that term means either.
